wiki/Introduction-to-Creating-a-Job.md

```diff
@@ -0,0 +1,1818 @@
+                -vf "scale=in_color_matrix=bt709:out_color_matrix=bt709" \
+                -frames:v 300 -c:v libx264 -preset fast \
+                -color_range tv -colorspace bt709 -color_primaries bt709 -color_trc iec61966-2-1 \
+                -movflags faststart "$OUTPUT_FILENAME"
+```
+
+As usual, you can check the syntax of the template and then try it out locally using the CLI:
+
+```bash
+% openjd check job.template.yaml                     
+Template at 'job.template.yaml' passes validation checks.
+
+% openjd run --step BlenderRender job.template.yaml \
+  -tp Frames=1..40 \
+  -p SceneFile=$(pwd)/3d/pavillon_barcelone_v1.2.blend \
+  -p FramesDirectory=$(pwd)/output_frames \
+  -p AnimationFile=$(pwd)/animation.mp4
+Mon Jul  8 10:48:30 2024	
+Mon Jul  8 10:48:30 2024	==============================================
+Mon Jul  8 10:48:30 2024	--------- Running Task
+Mon Jul  8 10:48:30 2024	==============================================
+Mon Jul  8 10:48:30 2024	Parameter values:
+Mon Jul  8 10:48:30 2024	Frames(STRING) = 1..40
+...
+```
+
+### 2.3. Ready for Production
+
+To develop the Job Template for this guide you've been reducing the resolution and quality of the resulting frames to make the
+development iterations quicker. To use the template in production you'd either need to remove the testing code from the `render.sh`
+embedded file so that it becomes:
+
+```bash
+#!/bin/bash
+
+# Return an error code if any command in the script fails.
+set -eou pipefail
+
+SCENE="$1"
+OUTDIR="$2"
+FRAME="$3"
+
+blender --background "$SCENE" \
+  --render-output "$OUTDIR"/frame-### \
+  --render-format PNG --use-extension 1 \
+  --render-frame "${FRAME}"
+```
+
+or you need to modify the template so that it has both a testing and production mode. We'll do the later in this guide to demonstrate
+using [Environments](2023-09-Template-Schemas#4-environment) in a Job Template. Think of Environments as encapsulating commands that you
+can run [before any Task has run, and after all Tasks have completed](How-Jobs-Are-Run#sessions) (whether successfully or not).
+Since they're just arbitrary commands of your choosing they can do anything that you would like, but one of their use cases is to
+set up or modify the environment where your Tasks will be running. For this case, modify the `render.sh` embedded file
+so that the testing code is only run if the environment variable `TESTING_TEMPLATE` has the value `true`:
+
+```bash
+#!/bin/bash
+
+# Return an error code if any command in the script fails.
+set -eou pipefail
+
+TMPFILE=$(mktemp)
+if test "${TESTING_TEMPLATE:-false}" == "true"; then
+    cat > "$TMPFILE" << EOF
+import bpy
+for s in bpy.data.scenes:
+s.render.resolution_x = 480
+s.render.resolution_y = 270
+bpy.context.scene.cycles.samples = 100
+EOF
+fi
+
+SCENE="$1"
+OUTDIR="$2"
+FRAME="$3"
+
+blender --background "$SCENE" \
+  --python "$TMPFILE" \
```

Comment:
this won't error with an empty file?

Reply:
Nope. Turns out that an empty file is a perfectly valid python script ;-)